File: backend/src/main/java/com/festago/school/dto/v1/SchoolFestivalArtistResponse.java
package com.festago.school.dto.v1;

import com.querydsl.core.annotations.QueryProjection;

public record SchoolFestivalArtistResponse(
    Long id,
    String name,
    String imageUrl
) {

    @QueryProjection
    public SchoolFestivalArtistResponse {
    }
}

File: android/festago/data/src/main/java/com/festago/festago/data/dto/schooldetail/SchoolFestivalArtistResponse.kt
package com.festago.festago.data.dto.schooldetail

import com.festago.festago.domain.model.artist.Artist
import kotlinx.serialization.Serializable

@Serializable
data class SchoolFestivalArtistResponse(
    val id: Int,
    val name: String,
    val profileImageUrl: String,
) {
    fun toDomain() = Artist(
        id = id.toLong(),
        name = name,
        imageUrl = profileImageUrl,
    )
}
